src/Factory.php

```diff
@@ -0,0 +1,74 @@
+<?php
+/**
+ * Part of the Joomla Framework Registry Package
+ *
+ * @copyright  Copyright (C) 2005 - 2015 Open Source Matters, Inc. All rights reserved.
+ * @license    GNU General Public License version 2 or later; see LICENSE
+ */
+
+namespace Joomla\Registry;
+
+/**
+ * Factory class to fetch Registry objects
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class Factory
+{
+	/**
+	 * Format instances container - for backward compatibility with AbstractRegistryFormat::getInstance().
+	 *
+	 * @var    array
+	 * @since  __DEPLOY_VERSION__
+	 * @deprecated  2.0  Object caching will no longer be supported
+	 */
+	protected static $formatInstances = array();
+
+	/**
+	 * Returns a AbstractRegistryFormat object, only creating it if it doesn't already exist.
+	 *
+	 * @param   string  $type     The format to load
+	 * @param   array   $options  Additional options to configure the object
+	 *
+	 * @return  AbstractRegistryFormat  Registry format handler
```

Comment:
It would make sense to me - because otherwise you start getting massive errors because your typehinting on some object that may be completely different. If we were only retrieving Joomla objects (where we know they implement the namespace) i think it's ok. But as soon as you could be loading any class then it should be typehinted